be/src/util/simd/bits.h

```diff
@@ -59,10 +59,28 @@ inline uint32_t bytes32_mask_to_bits32_mask(const bool* data) {
     return bytes32_mask_to_bits32_mask(reinterpret_cast<const uint8_t*>(data));
 }
 
-// compiler will make this SIMD automatically
 inline size_t count_zero_num(const int8_t* __restrict data, size_t size) {
     size_t num = 0;
     const int8_t* end = data + size;
+#if defined(__SSE2__) && defined(__POPCNT__)
+    const __m128i zero16 = _mm_setzero_si128();
+    const int8_t* end64 = data + (size / 64 * 64);
+
+    for (; data < end64; data += 64) {
+        num += __builtin_popcountll(
+                static_cast<uint64_t>(_mm_movemask_epi8(_mm_cmpeq_epi8(
```

Comment:
Maybe we can set num=size at start, and subtract the number of `1`?
In this way we can avoid using `_mm_cmpeq_epi8`.

Reply:
could you explain more? without _mm_cmpeq_epi8, how to figure out if the data is zero or not?

Reply:
we can shift lowest bit to highest, then movemask can get result.
```cpp
__builtin_popcountll(static_cast<uint64_t>(_mm_movemask_epi8(_mm_slli_epi16(
                                            *reinterpret_cast<const __m128i*>(data), 15))) |
                                    (static_cast<uint64_t>(_mm_movemask_epi8(_mm_slli_epi16(
                                             *reinterpret_cast<const __m128i*>(data + 16), 15)))
                                     << 16u) |
                                    (static_cast<uint64_t>(_mm_movemask_epi8(_mm_slli_epi16(
                                             *reinterpret_cast<const __m128i*>(data + 32), 15)))
                                     << 32u) |
                                    (static_cast<uint64_t>(_mm_movemask_epi8(_mm_slli_epi16(
                                             *reinterpret_cast<const __m128i*>(data + 48), 15)))
                                     << 48u));
```

Reply:
According to my test, it gives wrong answer.
I checked the intel docs for `_mm_slli_epi16`. Maybe the above codes need more intrinsics work together.

Here is the spec for `_mm_slli_epi16`:
FOR j := 0 to 7
	i := j*16
	IF imm8[7:0] > 15
		dst[i+15:i] := 0
	ELSE
		dst[i+15:i] := ZeroExtend16(a[i+15:i] << imm8[7:0])
	FI
ENDFOR

Reply:
Sorry, I find another way, we can shift bit directly.
But there doesn't seem to be a significant difference in performance either.

This should get the correct number of `1`:
```cpp
__builtin_popcountll(
                static_cast<uint64_t>(
                        _mm_movemask_epi8(*reinterpret_cast<const __m128i*>(data) << 7)) |
                (static_cast<uint64_t>(
                         _mm_movemask_epi8(*reinterpret_cast<const __m128i*>(data + 16) << 7))
                 << 16) |
                (static_cast<uint64_t>(
                         _mm_movemask_epi8(*reinterpret_cast<const __m128i*>(data + 32) << 7))
                 << 32) |
                (static_cast<uint64_t>(
                         _mm_movemask_epi8(*reinterpret_cast<const __m128i*>(data + 48) << 7))
                 << 48));
```